### PR TITLE
Simplify sequencer

### PR DESCRIPTION
**Pull request by pphaneuf**
### Checklist

- [ ] I have updated the [CHANGELOG](CHANGELOG.md).
- [ ] I have updated [documentation](docs/) accordingly (including the [feature implementation matrix](docs/Feature_Implementation_Matrix.md)).

### CODE REVIEW

File: log/sequencer.go [hunk 1]
```diff
@@ -292,17 +270,17 @@ func (s *preorderedLogSequencingTask) update(ctx context.Context, leaves []*tril
 
 // IntegrateBatch wraps up all the operations needed to take a batch of queued
 // or sequenced leaves and integrate them into the tree.
-func (s Sequencer) IntegrateBatch(ctx context.Context, tree *trillian.Tree, limit int, guardWindow, maxRootDurationInterval time.Duration) (int, error) {
-	start := s.timeSource.Now()
+func IntegrateBatch(ctx context.Context, tree *trillian.Tree, limit int, guardWindow, maxRootDurationInterval time.Duration, timeSource clock.TimeSource, logStorage storage.LogStorage, qm quota.Manager) (int, error) {
```
Comment (pavelkalinnikov) [inline log/sequencer.go:273]: Avoid name repetitions? Suggestion: `ts clock.TimeSource, ls storage.LogStorage`.

**Review by pavelkalinnikov — COMMENTED** [review]
Is this just a drive-by, or you have some bigger refactoring going? Honestly, I don't think small improvements can make this code significantly better, it needs a complete redesign.

This particular PR might be good, but feels like switching from C++ to C. I don't particularly like functions with many arguments of all kinds.

**Review by pavelkalinnikov — APPROVED** [review]
OTOH, currently this object is used "stateless", i.e. it's created, then the method is called once, and the object is forgotten. It makes some sense to just squash this whole thing to one function.

Approved modulo naming suggestion.

File: log/sequencer.go (hunk 1 above)
Comment (pphaneuf) [reply log/sequencer.go]: Done.

**Comment by pphaneuf** [comment]
Exactly, and having it as a struct with a method misleads someone reading the code into thinking that there might be cases where different code paths can happen.

This is some of the most mission-critical code in Trillian, "misleading" is the last thing we want!

**Pull request merged**